src/commands/edge_app.rs

```diff
@@ -234,6 +236,38 @@ impl EdgeAppCommand {
         Ok(())
     }
 
+    pub fn run(&self, path: &Path, secrets: Vec<(String, String)>) -> Result<(), anyhow::Error> {
+        let address_shared = Arc::new(Mutex::new(None));
+        let address_clone = address_shared.clone();
+
+        let runtime = tokio::runtime::Runtime::new().unwrap();
+        let path = path.to_path_buf();
+        runtime.block_on(async {
+            tokio::spawn(async move {
+                let address = run_server(path.as_path(), secrets).await.unwrap();
+                let mut locked_address = address_clone.lock().unwrap();
+                *locked_address = Some(address);
+            })
+            .await
+            .unwrap();
+
+            while address_shared.lock().unwrap().is_none() {
+                tokio::time::sleep(std::time::Duration::from_millis(100)).await;
+            }
+
+            println!(
+                "Edge app emulator is running at {}/index.html",
+                address_shared.lock().unwrap().as_ref().unwrap()
+            );
+
+            loop {
+                tokio::time::sleep(std::time::Duration::from_secs(3600)).await;
```

Comment:
Why just not forever?

Reply:
I did not find forever method

Reply:
I see there are some hacks for tokio, though they are not pretty(
https://stackoverflow.com/questions/66721384/how-can-i-make-my-program-sleep-for-a-long-time-while-staying-responsive

Reply:
No point in changing this then I think. Still won't look great.